src/Jeffijoe.MessageFormat/Formatting/BaseFormatter.cs

```diff
@@ -132,187 +140,198 @@ protected internal IEnumerable<KeyedBlock> ParseKeyedBlocks(FormatterRequest req
             const char EscapingChar = '\'';
 
             var result = new List<KeyedBlock>();
-            var key = new StringBuilder();
-            var block = new StringBuilder();
             var braceBalance = 0;
             var foundWhitespaceAfterKey = false;
             var insideEscapeSequence = false;
             if (request.FormatterArguments == null)
             {
                 return Enumerable.Empty<KeyedBlock>();
             }
-            
-            for (int i = startIndex; i < request.FormatterArguments.Length; i++)
-            {
-                var c = request.FormatterArguments[i];
-                var isWhitespace = char.IsWhiteSpace(c);
 
-                if (c == EscapingChar)
+            var key = StringBuilderPool.Get();
```

Comment:
Would it make sense to ensure we always get a `StringBuilder` with a certain capacity to avoid resizing? A key would never (sensibly) be more than 500 characters, for example, and I wonder if scanning ahead to the closing brace to figure out the size would be faster than potentially resizing?

Reply:
The Create overload does not give us the ability to choose capacity. I imagine that these pooled StringBuilders will extend to the size of the largest string the application is using for pluralization, and will remain at that size.

Reply:
Found out that any StringBuilders larger than 4k won't be preserved in the pool by default, they will be GCed away
See 
https://github.com/dotnet/aspnetcore/blob/main/src/ObjectPool/src/StringBuilderPooledObjectPolicy.cs#L35

Reply:
Which is reasonable, to avoid leaks